# Museum Data Analysis Project: Untangling The Mess
## Searching for derogatory terms

### Disclaimer: This work includes derogatory language that some people may find profane, vulgar, or offensive. The derogatory terms are used only for the purpose of finding them within museum collections. The inclusion of the offensive language is not meant to promote or condone any discriminatory or harmful beliefs. Reader discretion is advised.



In [1]:
# import libraries

import pandas as pd
import numpy as np
import requests
import warnings
import matplotlib.pyplot as plt
import seaborn as sns
import re

# set options
pd.set_option('display.max_columns', None)
pd.set_option('display.max_colwidth', 50)
import warnings 
  

warnings.filterwarnings('ignore') 

In [3]:
#load to dataframe
df = pd.read_csv('objects.csv')

dropped_cols = ['contextualtextcount', 'accesslevel', 'dateoflastpageview', 'totaluniquepageviews', 'markscount',
                'publicationcount', 'totaluniquepageviews', 'contact', 'colorcount', 'rank', 'id', 'state', 
                'verificationleveldescription', 'images','imagecount', 'worktypes', 'imagecount', 'classificationid',
                'standardreferencenumber', 'signed', 'relatedcount', 'verificationlevel', 'primaryimageurl', 'titlescount',
                'peoplecount', 'lastupdate', 'periodid', 'edition', 'lendingpermissionlevel', 'colors', 'groupcount',
                'url', 'dateoffirstpageview', 'datebegin', 'exhibitioncount', 'imagepermissionlevel', 'mediacount', 'techniqueid', 
                'seeAlso', 'details', 'videos', 'audio']

df.drop(dropped_cols, axis = 1, inplace = True)

## Pulling records with offensive keywords.
Here we can do an inital scan for some offensive keywords. Examples are as follows:
* squaw
* negro
* nigger
* savage
* eskimo
* indian
* redneck (red neck)
* jap
* oriental

In [5]:
def search_string_in_dataframe(df, search_string, case_sensitive=False):
    """
    Searches for a string in any field of a Pandas DataFrame.

    Args:
        df (pd.DataFrame): The DataFrame to search within.
        search_string (str): The string to search for.
        case_sensitive (bool, optional): Whether the search is case-sensitive. Defaults to False.

    Returns:
         pd.DataFrame: A DataFrame containing rows where the string was found, or an empty DataFrame if not found.
    """
    mask = np.column_stack([df[col].astype(str).str.contains(search_string, case=case_sensitive, na=False).values for col in df])
    return df[mask.any(axis=1)]

## Looking for "squaw" in records

In [7]:
squaw_df = search_string_in_dataframe(df, 'squaw')

In [8]:
print(squaw_df.shape)
pd.set_option('display.max_columns', None)
pd.set_option('display.max_colwidth', 150)
display(squaw_df)
print(squaw_df.title)

(3, 27)


Unnamed: 0 copyright                                       creditline  \
67679          79       NaN  Harvard Art Museums/Fogg Museum, Anonymous Gift   
68164          64       NaN  Harvard Art Museums/Fogg Museum, Anonymous Gift   
68734          34       NaN  Harvard Art Museums/Fogg Museum, Anonymous Gift   

                      createdate                   division period  \
67679  2002-02-16T00:00:00-05:00  European and American Art    NaN   
68164  2002-02-16T00:00:00-05:00  European and American Art    NaN   
68734  2002-02-06T00:00:00-05:00  European and American Art    NaN   

       totalpageviews  accessionyear classification style commentary  \
67679              31         1953.0       Drawings   NaN        NaN   
68164              17         1953.0       Drawings   NaN        NaN   
68734              20         1953.0       Drawings   NaN        NaN   

      technique description  \
67679       NaN         NaN   
68164       NaN         NaN   
68734       NaN         NaN   

                                                                                                                      medium  \
67679  Watercolor and black ink over traces of graphite on off-white antique laid paper, adhered to cream antique laid paper   
68164  Watercolor and black ink over traces of graphite on off-white antique laid paper, adhered to cream antique laid paper   
68734  Watercolor and black ink over traces of graphite on off-white antique laid paper, adhered to cream antique laid paper   

                                                                title  \
67679  Squawking Bird with Talons Extended to the Right; verso: blank   
68164               Two Perched and Squawking Pheasants; verso: blank   
68734                  Two Squawking Ducks in the Water; verso: blank   

      accessionmethod provenance         dated              department  \
67679            Gift        NaN  18th century  Department of Drawings   
68164            Gift        NaN  18th century  Department of Drawings   
68734            Gift        NaN  18th century  Department of Drawings   

       dateend  \
67679        0   
68164        0   
68734        0   

                                                                                                                                                      people  \
67679  [{'role': 'Artist', 'birthplace': 'Paris', 'gender': 'male', 'displaydate': '1686 - 1755', 'prefix': None, 'culture': 'French', 'displayname': 'Je...   
68164  [{'role': 'Artist', 'birthplace': 'Paris', 'gender': 'male', 'displaydate': '1686 - 1755', 'prefix': None, 'culture': 'French', 'displayname': 'Je...   
68734  [{'role': 'Artist', 'birthplace': 'Paris', 'gender': 'male', 'displaydate': '1686 - 1755', 'prefix': None, 'culture': 'French', 'displayname': 'Je...   

            century objectnumber labeltext culture  objectid  \
67679  18th century   1953.75.38       NaN  French    147157   
68164  18th century   1953.75.23       NaN  French    147706   
68734  18th century   1953.75.52       NaN  French    148457   

                                             dimensions  
67679                                               NaN  
68164  irregular: 33.2 x 30 cm (13 1/16 x 11 13/16 in.)  
68734                                               NaN

67679    Squawking Bird with Talons Extended to the Right; verso: blank
68164                 Two Perched and Squawking Pheasants; verso: blank
68734                    Two Squawking Ducks in the Water; verso: blank
Name: title, dtype: object


## Squaw returned three records with 'squaw'king birds in tile field

## Now searching for 'negro' in records

In [89]:
ne_df = search_string_in_dataframe(df, 'negro')
ne_df.to_csv('ne.csv')

In [85]:
print(f"There are {ne_df.shape[0]} entries with negro in a field.")
pd.set_option('display.max_columns', None)
pd.set_option('display.max_colwidth', None)
display(ne_df[:10])


There are 306 entries with negro in a field.


Unnamed: 0 copyright  \
415          15       NaN   
416          16       NaN   
417          17       NaN   
674          74       NaN   
675          75       NaN   
676          76       NaN   
677          77       NaN   
678          78       NaN   
765          65       NaN   
766          66       NaN   

                                                                                                            creditline  \
415  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
416  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
417  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
674  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
675  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
676  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
677  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
678  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
765  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
766  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   

                    createdate                     division period  \
415  2004-03-11T00:00:00-05:00  Modern and Contemporary Art    NaN   
416  2004-06-02T00:00:00-04:00  Modern and Contemporary Art    NaN   
417  2004-03-11T00:00:00-05:00  Modern and Contemporary Art    NaN   
674  2004-07-14T00:00:00-04:00  Modern and Contemporary Art    NaN   
675  2004-07-14T00:00:00-04:00  Modern and Contemporary Art    NaN   
676  2004-06-02T00:00:00-04:00  Modern and Contemporary Art    NaN   
677  2004-03-11T00:00:00-05:00  Modern and Contemporary Art    NaN   
678  2004-03-11T00:00:00-05:00  Modern and Contemporary Art    NaN   
765  2004-03-11T00:00:00-05:00  Modern and Contemporary Art    NaN   
766  2004-03-11T00:00:00-05:00  Modern and Contemporary Art    NaN   

     totalpageviews  accessionyear classification style commentary  \
415              58         2011.0    Photographs   NaN        NaN   
416              91         2011.0    Photographs   NaN        NaN   
417              37         2011.0    Photographs   NaN        NaN   
674              18         2011.0    Photographs   NaN        NaN   
675              24         2011.0    Photographs   NaN        NaN   
676              34         2011.0    Photographs   NaN        NaN   
677              13         2011.0    Photographs   NaN        NaN   
678              42         2011.0    Photographs   NaN        NaN   
765              21         2011.0    Photographs   NaN        NaN   
766              18         2011.0    Photographs   NaN        NaN   

          technique description                                       medium  \
415  Platinum print         NaN                                          NaN   
416  Platinum print         NaN                                          NaN   
417  Platinum print         NaN  Platinum prints mounted to board, black ink   
674  Platinum print         NaN                                          NaN   
675  Platinum print         NaN                                          NaN   
676  Platinum print         NaN                                          NaN   
677  Platinum print         NaN                                          NaN   
678  Platinum print         NaN                                          NaN   
765  Platinum print         NaN                                          NaN   


## Negro was found in 306 entries, in 297 the word was in the title.

In [77]:
title_ne_df = search_string_in_dataframe(pd.DataFrame(ne_df['title']), 'negro')
print(f"There are {len(title_ne_df)} records with the word negro in the title.")

There are 297 records with the word negro in the title.


In [75]:
display(title_ne_df.title.unique())

array(['Races, Negroes: United States. Virginia. Hampton. Hampton Normal and Industrial School: Agencies Promoting Assimilation of the Negro: Hampton Normal and Agricultural Institute, Hampton, Va.',
       'Races, Negroes: United States. Virginia. Hampton. Hampton Normal and Industrial School: Agencies Promoting Assimilation of the Negro. Training for Commercial and Industrial Employment. Hampton Normal and Agricultural Institute, Hampton, Va.',
       'Races, Negroes: United States. Virginia. Hampton. Hampton Normal and Industrial School: Agencies Promoting Assimilation of the Negro. Hampton Normal and Agricultural Institute, Hampton, Va.',
       'Races, Negroes: United States. Virginia. Hampton. Hampton Normal and Industrial School: Agencies Promoting Assimilation of the Negro. Development of Social Standards Among the Negroes. Hampton Normal and Agricultural Institute, Hampton, Va.: Arithmetic Class measuring brick.',
       'Races, Negroes: United States. Virginia. Hampton. Hampt

In [87]:
not_title_ne_df = ne_df.merge(title_ne_df, how='left', indicator=True).query('_merge == "left_only"').drop(columns=['_merge'])
display(not_title_ne_df)

Unnamed: 0  \
85            6   
148          51   
410           5   
431          21   
432          16   
499          35   
500          40   
505           8   
507          51   

                                                                                            copyright  \
85                                                                                                NaN   
148  ©  The Jacob and Gwendolyn Lawrence Foundation, Seattle / Artists Rights Society (ARS), New York   
410                                                                                               NaN   
431                                                                                               NaN   
432                                                                                               NaN   
499                                 © John T. Biggers Estate / Artists Rights Society (ARS), New York   
500                                                                                       © Roni Horn   
505                                                                                               NaN   
507                                                                                               NaN   

                                                                                                                                             creditline  \
85                                                                           Harvard Art Museums/Fogg Museum, Gift of Sarah-Ann and Werner H. Kramarsky   
148                    Harvard Art Museums/Fogg Museum, Gift of Lorillard, Division of Loews Theatres, Inc., transfer from student print rental program   
410                                           Harvard Art Museums/Fogg Museum, Gift of the artist and Martha Tepper Takayama, Tepper Takayama Fine Arts   
431                         Harvard Art Museums/Arthur M. Sackler Museum, Loan from the Trustees of the Arthur Stone Dewing Greek Numismatic Foundation   
432                                                                          Harvard Art Museums/Arthur M. Sackler Museum, Bequest of David M. Robinson   
499                                                                       Harvard Art Museums/Fogg Museum, Bequest of William S. Lieberman, by exchange   
500                                                                                       Harvard Art Museums/Fogg Museum, Richard Norton Memorial Fund   
505                                Harvard Art Museums/Straus Center for Conservation and Technical Studies, Alan Burroughs Collection of X-Radiographs   
507  Harvard Art Museums/Fogg Museum, Purchase through the generosity of Anne H. and Frederick Vogel III and bequest of Nettie G. Naumburg, by exchange   

                    createdate                     division  \
85   2005-11-17T00:00:00-05:00  Modern and Contemporary Art   
148  2002-08-08T00:00:00-04:00  Modern and Contemporary Art   
410  2001-11-29T00:00:00-05:00  Modern and Contemporary Art   
431  1999-12-09T00:00:00-05:00  Asian and Mediterranean Art   
432  1997-07-06T00:00:00-04:00  Asian and Mediterranean Art   
499  2008-02-04T15:56:43-05:00  Modern and Contemporary Art   
500  2008-05-20T14:41:26-04:00  Modern and Contemporary Art   
505  2013-02-27T14:02:40-05:00                          NaN   
507  2017-03-29T11:51:04-04:00    European and American Art   

                       period  totalpageviews  accessionyear classification  \
85                        NaN               8         2005.0       Drawings   
148                       NaN            1866         2005.0         Prints   
410                       NaN              14         2001.0    Photographs   
431  Hellenistic period, Late              50            NaN          Coins   
432   Classical period, Early             254         1960.0        Vessels   
499                       NaN             483         2008.0         Prints   
500                       Na

## Searching for word n****r

In [97]:
ni_df = search_string_in_dataframe(df, 'nigger')
ni_df.to_csv('ni.csv')

In [101]:
print(f"There are {ni_df.shape[0]} entries with n****r in a field.")
display(ni_df)

There are 3 entries with n****r in a field.


Unnamed: 0                                                  copyright  \
8838            38                                                        NaN   
155412          12  © Faith Ringgold / Artists Rights Society (ARS), New York   
211961          61                                                        NaN   

                                                                                                           creditline  \
8838    Harvard Art Museums/Fogg Museum, On loan from Special Collections, Fine Arts Library, Harvard College Library   
155412                                                  Harvard Art Museums/Fogg Museum, Richard Norton Memorial Fund   
211961                                                 Harvard Art Museums/Fogg Museum, Francis H. Burr Memorial Fund   

                       createdate                     division period  \
8838    2005-08-31T00:00:00-04:00    European and American Art    NaN   
155412  2013-10-30T12:15:35-04:00  Modern and Contemporary Art    NaN   
211961  1997-10-21T00:00:00-04:00  Modern and Contemporary Art    NaN   

        totalpageviews  accessionyear classification style commentary  \
8838               567            NaN         Prints   NaN        NaN   
155412            3665         2014.0      Paintings   NaN        NaN   
211961              28         1955.0         Prints   NaN        NaN   

               technique description         medium  \
8838    Chromolithograph         NaN            NaN   
155412               NaN         NaN  Oil on canvas   
211961          Drypoint         NaN            NaN   

                                                title accessionmethod  \
8838                                      Free Nigger    Not Recorded   
155412  Black Light Series #8: Red White Black Nigger        Purchase   
211961                                     Niggertanz        Purchase   

                                                                                                                                                  provenance  \
8838                                                                                                                                                     NaN   
155412                                               Faith Ringgold, created 1969, sold [through ACA Galleries, New York]; to the Harvard Art Museums, 2014.   
211961  Paul A. Struck, sold; to Fogg Art Museum, December 27, 1955.\r\n\r\nNOTE: This is possibly from Paul A. Struck, Inc., Publisher, New York, New York.   

               dated                               department  dateend  \
8838    19th century                     Department of Prints        0   
155412          1969  Department of Modern & Contemporary Art     1969   
211961  20th century                     Department of Prints        0   

                                                                                                                                                                                                                                                                                                                           people  \
8838                          [{'role': 'Artist', 'birthplace': None, 'gender': 'unknown', 'displaydate': '1810 - 1870', 'prefix': None, 'culture': 'American', 'displayname': 'John Henry Bufford', 'alphasort': 'Bufford, John Henry', 'name': 'John Henry Bufford', 'personid': 52989, 'deathplace': None, 'displayorder': 1}]   
155412  [{'role': 'Artist', 'birthplace': 'New York, New York', 'gender': 'female', 'displaydate': '1930 - 2024', 'prefix': None, 'culture': 'American', 'displayname': 'Faith Ringgold', 'alphasort': 'Ringgold, Faith', 'name': 'Faith Ringgold', 'personid': 18532, 'deathplace': 'Englewood, New Jersey', 'displayorder': 1}]   
211961                [{'role': 'Artist', 'birthplace': 'Leipzig, Germany', 'gender': 'male', 'displaydate': '1884 - 1950', 'prefix': None, 'culture': 'German', 'displayname': 'Max Beck

## Of three results, all have the keyword in the title, one is a contemporary work by black woman, one is in German and one is a 19th century print

## Searching for word savage

In [138]:
sa_df = search_string_in_dataframe(df, 'savage')
sa_df.to_csv('savage.csv')

In [140]:
print(f"There are {sa_df.shape[0]} entries with savage in a field.")
display(sa_df)

There are 35 entries with savage in a field.


Unnamed: 0  \
73014           14   
75379           79   
75797           97   
75809            9   
113562          62   
121101           1   
123172          72   
123404           4   
128349          49   
131964          64   
132051          51   
132060          60   
132076          76   
134005           5   
147722          22   
148452          52   
149678          78   
153500           0   
153755          55   
153918          18   
156518          18   
157016          16   
183885          85   
202317          17   
214926          26   
216286          86   
216520          20   
216850          50   
220484          84   
220691          91   
220692          92   
220900           0   
222663          63   
232186          86   
235178          78   

                                                         copyright  \
73014                                                          NaN   
75379                                                          NaN   
75797                                                          NaN   
75809                                                          NaN   
113562                                                         NaN   
121101                                                         NaN   
123172                                                         NaN   
123404                                                         NaN   
128349                                                         NaN   
131964                                                         NaN   
132051                                                         NaN   
132060                                                         NaN   
132076                                                         NaN   
134005                                                         NaN   
147722                                                         NaN   
148452  © Kara Walker, Courtesy of Sikkema Jenkins & Co., New York   
149678                               © Gilles Peress/Magnum Photos   
153500                                                         NaN   
153755                                                         NaN   
153918                                                         NaN   
156518                                                         NaN   
157016                                                         NaN   
183885                                                         NaN   
202317                                                         NaN   
214926                                                         NaN   
216286                                                         NaN   
216520                                                         NaN   
216850                                                         NaN   
220484                                                         NaN   
220691                                                         NaN   
220692                                                         NaN   
220900                                                         NaN   
222663                                                         NaN   
232186                                                         NaN   
235178                                                         NaN   

                                                                                                                                                  creditline  \
73014                                                                Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts   
75379   Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Transferred from the Gardner Collection, Geology Department   
75797   Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Transferred from the Gardner Collection, Geology Department   
75809   Harvard Art Museums/Fogg Museum, Transfer from

## Many of the records have savage as a last name. So I will exclude those from search

In [154]:
sa_df.drop(['provenance', 'people', 'creditline'], axis = 1, inplace = True)

KeyError: "['provenance', 'people', 'creditline'] not found in axis"

In [150]:
sa_df = search_string_in_dataframe(sa_df, 'savage')

In [158]:
sa_df.shape

(18, 24)

In [156]:
display(sa_df)

Unnamed: 0  \
113562          62   
128349          49   
134005           5   
147722          22   
148452          52   
149678          78   
153500           0   
153755          55   
153918          18   
183885          85   
216286          86   
216520          20   
216850          50   
220484          84   
220691          91   
220692          92   
220900           0   
222663          63   

                                                         copyright  \
113562                                                         NaN   
128349                                                         NaN   
134005                                                         NaN   
147722                                                         NaN   
148452  © Kara Walker, Courtesy of Sikkema Jenkins & Co., New York   
149678                               © Gilles Peress/Magnum Photos   
153500                                                         NaN   
153755                                                         NaN   
153918                                                         NaN   
183885                                                         NaN   
216286                                                         NaN   
216520                                                         NaN   
216850                                                         NaN   
220484                                                         NaN   
220691                                                         NaN   
220692                                                         NaN   
220900                                                         NaN   
222663                                                         NaN   

                       createdate                     division  \
113562  1997-07-06T00:00:00-04:00  Asian and Mediterranean Art   
128349  1999-01-20T00:00:00-05:00    European and American Art   
134005  2006-03-27T00:00:00-05:00    European and American Art   
147722  2009-10-26T11:51:19-04:00    European and American Art   
148452  2011-06-09T11:26:18-04:00  Modern and Contemporary Art   
149678  2011-11-17T10:51:43-05:00  Modern and Contemporary Art   
153500  2013-02-27T13:47:13-05:00                          NaN   
153755  2013-02-27T13:47:33-05:00                          NaN   
153918  2013-02-27T13:51:12-05:00                          NaN   
183885  1997-11-01T00:00:00-05:00    European and American Art   
216286  1997-10-19T00:00:00-04:00    European and American Art   
216520  1997-10-19T00:00:00-04:00    European and American Art   
216850  1997-10-19T00:00:00-04:00    European and American Art   
220484  1997-10-18T00:00:00-04:00    European and American Art   
220691  1997-10-18T00:00:00-04:00    European and American Art   
220692  1997-10-18T00:00:00-04:00    European and American Art   
220900  1997-10-18T00:00:00-04:00    European and American Art   
222663  1997-10-18T00:00:00-04:00    European and American Art   

                period  totalpageviews  accessionyear classification  \
113562  Archaic period            1968         1927.0        Vessels   
128349             NaN             173            NaN      Paintings   
134005             NaN              27            NaN      Paintings   
147722             NaN             167         2009.0         Prints   
148452             NaN             458         2011.0         Prints   
149678             NaN             335         2011.0    Photographs   
153500             NaN               0            NaN    Photographs   
153755             NaN               7            NaN    Photographs   
153918             NaN              10            NaN    Photographs   
183885             NaN            7051         1943.0      Paintings   
216286             NaN               5         1992.0         Prints   
216520             NaN               3         1992.0         Prints   
216850             NaN               0         1992.0         Print

In [160]:
sa_df.to_csv('savage.csv')

## There are a total of 18 records with the keyword 'savage'. A few are using the term savage as a description for a person or peoples.

## Searching for keyword 'eskimo'

In [164]:
esk_df = search_string_in_dataframe(df, 'eskimo')

In [166]:
esk_df.shape

(0, 27)

## There are no records containing keyword 'eskimo'

## Searching for keyword 'indian'

In [173]:
ind_df = search_string_in_dataframe(df, 'indian')

In [175]:
ind_df.shape

(4865, 27)

In [179]:
display(ind_df[:5])

Unnamed: 0 copyright  \
13           13       NaN   
92           92       NaN   
148          48       NaN   
151          51       NaN   
152          52       NaN   

                                                                                                                                                                                                                                                                                                                                              creditline  \
13   Harvard Art Museums/Arthur M. Sackler Museum, Gift in gratitude to John Coolidge, Gift of Leslie Cheek, Jr., Anonymous Fund in memory of Henry Berg, Louise Haskell Daly, Alpheus Hyatt, Richard Norton Memorial Funds and through the generosity of Albert H. Gordon and Emily Rauh Pulitzer; formerly in the collection of Stuart Cary Welch, Jr.   
92                                                                                                                                                                                                                          Harvard Art Museums/Arthur M. Sackler Museum, Grace Nichols Strong, Francis H. Burr and Friends of the Fogg Art Museum Funds   
148                                                                                                                                                                                                                                                                         Harvard Art Museums/Arthur M. Sackler Museum, Gift of Stuart Cary Welch, Jr.   
151                                                                                                                                                                                                                                 Harvard Art Museums/Arthur M. Sackler Museum, Gweneth Knight Memorial Fund and the William M. Prichard Memorial Fund   
152                                                                                                                                                                                                                                                                             Harvard Art Museums/Arthur M. Sackler Museum, Gift of Victoria S. Munroe   

                    createdate                     division          period  \
13   2006-03-24T00:00:00-05:00  Asian and Mediterranean Art             NaN   
92   2006-01-17T00:00:00-05:00  Asian and Mediterranean Art  Safavid period   
148  2006-01-24T00:00:00-05:00  Asian and Mediterranean Art             NaN   
151  2006-03-08T00:00:00-05:00  Asian and Mediterranean Art             NaN   
152  2006-03-08T00:00:00-05:00  Asian and Mediterranean Art             NaN   

     totalpageviews  accessionyear classification style  \
13             1237         1995.0    Manuscripts   NaN   
92              592         1950.0         Albums   NaN   
148              60         2006.0   Textile Arts   NaN   
151             219         2006.0       Drawings   NaN   
152              43         2006.0       Drawings   NaN   

                                                                                                                                                                                                                                                                                                                                                                                             commentary  \
13                                                                                                                                                                                                                                                                                                                                                                                                  NaN   
92                                                                                                                       

## This search returned 4865 records. We want to look at uses of the word 'indian' not realated to India, so to start we will eliminate the records within the division of 'Asian and Mediterranean Art', records that have 'Indian' for culture and records with the word 'Indiana'

In [210]:
ind_df = ind_df[ind_df['division'] !='Asian and Mediterranean Art']
ind_df = ind_df[ind_df['culture'] !='Indian']
indiana_ind_df = search_string_in_dataframe(ind_df, 'Indiana')
ind_df = ind_df.merge(indiana_ind_df, how='left', indicator=True).query('_merge == "left_only"').drop(columns=['_merge'])
ind_df.shape

(271, 27)

In [203]:
display(ind_df[:10])
ind_df.to_csv('indian.csv')

Unnamed: 0 copyright  \
1843          43       NaN   
1844          44       NaN   
2032          32       NaN   
2106           6       NaN   
2107           7       NaN   
2108           8       NaN   
2134          34       NaN   
2135          35       NaN   
2370          70       NaN   
2371          71       NaN   

                                                                                                             creditline  \
1843  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
1844  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
2032  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
2106  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
2107  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
2108  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
2134  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
2135  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
2370  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
2371  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   

                     createdate                     division period  \
1843  2006-02-10T00:00:00-05:00  Modern and Contemporary Art    NaN   
1844  2006-02-10T00:00:00-05:00  Modern and Contemporary Art    NaN   
2032  2006-02-10T00:00:00-05:00  Modern and Contemporary Art    NaN   
2106  2006-02-10T00:00:00-05:00  Modern and Contemporary Art    NaN   
2107  2006-02-10T00:00:00-05:00  Modern and Contemporary Art    NaN   
2108  2006-02-10T00:00:00-05:00  Modern and Contemporary Art    NaN   
2134  2006-02-17T00:00:00-05:00  Modern and Contemporary Art    NaN   
2135  2006-02-17T00:00:00-05:00  Modern and Contemporary Art    NaN   
2370  2006-02-10T00:00:00-05:00  Modern and Contemporary Art    NaN   
2371  2006-02-10T00:00:00-05:00  Modern and Contemporary Art    NaN   

      totalpageviews  accessionyear classification style commentary  \
1843              13         2011.0    Photographs   NaN        NaN   
1844               9         2011.0    Photographs   NaN        NaN   
2032               5         2011.0    Photographs   NaN        NaN   
2106               9         2011.0    Photographs   NaN        NaN   
2107              16         2011.0    Photographs   NaN        NaN   
2108               4         2011.0    Photographs   NaN        NaN   
2134               5         2011.0    Photographs   NaN        NaN   
2135               2         2011.0    Photographs   NaN        NaN   
2370               3         2011.0    Photographs   NaN        NaN   
2371               5         2011.0    Photographs   NaN        NaN   

                 technique description medium  \
1843  Gelatin silver print         NaN    NaN   
1844  Gelatin silver print         NaN    NaN   
2032  Gelatin silver print         NaN    NaN   
2106  Gelatin silver print         NaN    NaN   
2107  Gelatin silver print         NaN    NaN   
2108  Gelatin silver print         NaN    NaN   
2134        Platinum print         NaN    NaN   
2135        Platinum print         NaN    NaN   
2370  Gelatin silver print         NaN    NaN   
2371  Gelatin silver print         NaN    NaN   

                                                                                                                                                                                                                         

## We were able to get the count of records down to 271. The first 20 that we can see are photographs with titles in the style of:
"Races, Indians: United States. New York. Iroquois. Thomas Asylum for Orphan and Destitute Indian Children: State Thomas Asylum for Orphan and Destitute Indian Children, Iroquois, N.Y.: Laundry"

In [214]:
races_ind_df= search_string_in_dataframe(ind_df, 'Races, Indians')
races_ind_df.shape

(45, 27)

In [218]:
not_races_ind_df = ind_df.merge(races_ind_df, how='left', indicator=True).query('_merge == "left_only"').drop(columns=['_merge'])
display(not_races_ind_df[:20])

Unnamed: 0                                copyright  \
6           34                                      NaN   
7           35                                      NaN   
26          15                                      NaN   
28          66                                      NaN   
31          68                                      NaN   
32          27                                      NaN   
33          34  © 1991 Harry Sternberg / Brighton Press   
34          12      © Estate of Joseph Janney Steinmetz   
35          88                                      NaN   
36          58                                      NaN   
37          85                                      NaN   
38          46                                      NaN   
39          21                                      NaN   
40          17                                      NaN   
41          90      © Estate of Joseph Janney Steinmetz   
42          91      © Estate of Joseph Janney Steinmetz   
43          71                                      NaN   
44          46                                      NaN   
45           6                 © Estate of Louis Faurer   
46          54                                      NaN   

                                                                                                                                 creditline  \
6                         Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
7                         Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
26                                                                                 Harvard Art Museums/Fogg Museum, Gift of Jose M. Soriano   
28                        Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, Social Museum Collection   
31                                                                             Harvard Art Museums/Fogg Museum, Acquisition Fund for Prints   
32                                                                             Harvard Art Museums/Fogg Museum, Acquisition Fund for Prints   
33                                                                                    Harvard Art Museums/Fogg Museum, Margaret Fisher Fund   
34  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   
35                                                                            Harvard Art Museums/Fogg Museum, Gift of Richard Stuart Teitz   
36  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   
37  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   
38  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   
39  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   
40  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   
41  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   
42  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   
43  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   
44  Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   
45                                             

## The list now contains numerous examples of Indian and 'American Indian' refering to indigenous american peoples.

In [221]:
red_df = search_string_in_dataframe(df, 'redneck')

In [227]:
red_df.shape

(0, 27)

In [229]:
red_df = search_string_in_dataframe(df, 'red neck')

In [231]:
red_df.shape

(10, 27)

In [233]:
display(red_df)

Unnamed: 0 copyright  \
36697           97       NaN   
83823           23       NaN   
97040           40       NaN   
97048           48       NaN   
97073           73       NaN   
102593          93       NaN   
108413          13       NaN   
115306           6       NaN   
116758          58       NaN   
116983          83       NaN   

                                                                                                                                                                                creditline  \
36697                                                      Harvard Art Museums/Arthur M. Sackler Museum, Transfer from the Peabody Museum of Archaeology and Ethnology, Harvard University   
83823                                                 Harvard Art Museums/Arthur M. Sackler Museum, Francis H. Burr Memorial Fund and the Discretionary Fund of the Islamic Art Department   
97040                       Harvard Art Museums/Arthur M. Sackler Museum, Partial gift of the Walter C. Sedgwick Foundation and partial purchase through the Alpheus Hyatt Purchasing Fund   
97048   Harvard Art Museums/Arthur M. Sackler Museum, Partial gift of the Walter C. Sedgwick Foundation and partial purchase through the Ernest B. and Helen Pratt Dane Fund for Asian Art   
97073   Harvard Art Museums/Arthur M. Sackler Museum, Partial gift of the Walter C. Sedgwick Foundation and partial purchase through the Ernest B. and Helen Pratt Dane Fund for Asian Art   
102593  Harvard Art Museums/Arthur M. Sackler Museum, Partial gift of the Walter C. Sedgwick Foundation and partial purchase through the Ernest B. and Helen Pratt Dane Fund for Asian Art   
108413  Harvard Art Museums/Arthur M. Sackler Museum, Partial gift of the Walter C. Sedgwick Foundation and partial purchase through the Ernest B. and Helen Pratt Dane Fund for Asian Art   
115306                                                                                                               Harvard Art Museums/Arthur M. Sackler Museum, Gift of Stuart C. Welch   
116758                                                                                         Harvard Art Museums/Arthur M. Sackler Museum, Harvard-Baghdad School Expedition (1929-1930)   
116983                                                                                         Harvard Art Museums/Arthur M. Sackler Museum, Harvard-Baghdad School Expedition (1929-1930)   

                       createdate                     division  \
36697   2003-09-05T00:00:00-04:00  Asian and Mediterranean Art   
83823   2002-03-20T00:00:00-05:00  Asian and Mediterranean Art   
97040   2000-04-17T00:00:00-04:00  Asian and Mediterranean Art   
97048   2000-04-14T00:00:00-04:00  Asian and Mediterranean Art   
97073   2000-04-17T00:00:00-04:00  Asian and Mediterranean Art   
102593  1999-09-24T00:00:00-04:00  Asian and Mediterranean Art   
108413  1998-10-16T00:00:00-04:00  Asian and Mediterranean Art   
115306  1997-07-06T00:00:00-04:00  Asian and Mediterranean Art   
116758  2002-03-28T00:00:00-05:00  Asian and Mediterranean Art   
116983  1997-07-06T00:00:00-04:00  Asian and Mediterranean Art   

                                                               period  \
36697                                                             NaN   
83823                                                   Mamluk period   
97040            Neolithic period, Dawenkou culture, c. 4300-2600 BCE   
97048                                           Tang dynasty, 618-907   
97073           Neolithic period, Peiligang culture, c. 6500-5000 BCE   
102593           Neolithic period, Dawenkou culture, c. 4300-2600 BCE   
108413  Neolithic period, Shandong Longshan culture, c. 2600-2000 BCE   
115306                                             Bronze Age, Middle   
116758                                               Mitannian period   
116983                                                Sasanian period   

        totalpageviews  accession

## Neither 'redneck' nor 'red neck' returned any records with derogatory language  

# Searching for keyword'jap'

In [236]:
j_df = search_string_in_dataframe(df, ' jap ')

In [238]:
j_df.shape

(0, 27)

## keyword 'jap' returned no results

# Searching for keyword oriental

In [275]:
o_df = search_string_in_dataframe(df, 'oriental')

In [277]:
o_df.shape

(281, 27)

In [252]:
display(o_df[:10])

Unnamed: 0                            copyright  \
140            40                                  NaN   
6409            9                                  NaN   
29267          67                                  NaN   
29319          19                                  NaN   
30689          89                                  NaN   
30721          21                                  NaN   
40165          65                                  NaN   
40166          66                                  NaN   
40210          10                                  NaN   
40898          98  © Estate of Joseph Janney Steinmetz   

                                                                                                                                              creditline  \
140                          Harvard Art Museums/Arthur M. Sackler Museum, Gift of Arielle and Jerry Brodkey (Class of 1955) in memory of Leo Mildenberg   
6409                                                                                Harvard Art Museums/Arthur M. Sackler Museum, Gift of Prentice Duell   
29267                                                       Harvard Art Museums/Arthur M. Sackler Museum, Gift of Shirley Nye in memory of Warren E. Cox   
29319                                                                Harvard Art Museums/Arthur M. Sackler Museum, Gift of Benjamin and Lilian Hertzberg   
30689  Harvard Art Museums/Arthur M. Sackler Museum, Transfer from the Alice Corinne McDaniel Collection, Department of the Classics, Harvard University   
30721                                                                Harvard Art Museums/Arthur M. Sackler Museum, Gift of Benjamin and Lilian Hertzberg   
40165                             Harvard Art Museums/Arthur M. Sackler Museum, Purchase through the generosity of Claude Grenier in honor of James Cuno   
40166                                                                               Harvard Art Museums/Arthur M. Sackler Museum, David M. Robinson Fund   
40210                                                                               Harvard Art Museums/Arthur M. Sackler Museum, David M. Robinson Fund   
40898            Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   

                      createdate                     division  \
140    2004-12-22T00:00:00-05:00  Asian and Mediterranean Art   
6409   2006-01-04T00:00:00-05:00  Asian and Mediterranean Art   
29267  2004-04-14T00:00:00-04:00  Asian and Mediterranean Art   
29319  2004-06-01T00:00:00-04:00  Asian and Mediterranean Art   
30689  2004-06-15T00:00:00-04:00  Asian and Mediterranean Art   
30721  2004-06-01T00:00:00-04:00  Asian and Mediterranean Art   
40165  2002-12-18T00:00:00-05:00  Asian and Mediterranean Art   
40166  2002-12-18T00:00:00-05:00  Asian and Mediterranean Art   
40210  2002-12-18T00:00:00-05:00  Asian and Mediterranean Art   
40898  2003-01-07T00:00:00-05:00  Modern and Contemporary Art   

                                             period  totalpageviews  \
140                            Orientalizing period             132   
6409                            Neo-Assyrian period              73   
29267  Song dynasty, Northern Song period, 960-1127              19   
29319                          Orientalizing period              23   
30689                          Orientalizing period              36   
30721                          Orientalizing period              24   
40165                          Orientalizing period            2012   
40166                          Orientalizing period               5   
40210                          Orientalizing period              59   
40898                                           NaN               2   

       accessionyear classification       style  \
140           2004.0        Jewelry         NaN   
6409          1942.0      Fragments         NaN   


## 'Orientalizing period' is a common value in period field

In [279]:
n_o_df =o_df[~o_df['period'].isna()]
n_o_df = n_o_df[n_o_df['period'].str.contains('Orientalizing')]
n_o_df.shape

(124, 27)

In [281]:
o_df = o_df.merge(n_o_df, how='left', indicator=True).query('_merge == "left_only"').drop(columns=['_merge'])
print(o_df.shape)
display(o_df[:20])

(157, 27)


Unnamed: 0                            copyright  \
1            9                                  NaN   
2           67                                  NaN   
9           98  © Estate of Joseph Janney Steinmetz   
14          18  © Estate of Joseph Janney Steinmetz   
15          14                                  NaN   
16          31                                  NaN   
26          28                                  NaN   
27          29                                  NaN   
28          62                                  NaN   
34          83                                  NaN   
35          38                                  NaN   
36          41                                  NaN   
37          42                                  NaN   
38          43                                  NaN   
39          45                                  NaN   
40          46                                  NaN   
41          47                                  NaN   
42          63                                  NaN   
43          66                                  NaN   
44          10                                  NaN   

                                                                                                                                                        creditline  \
1                                                                                             Harvard Art Museums/Arthur M. Sackler Museum, Gift of Prentice Duell   
2                                                                     Harvard Art Museums/Arthur M. Sackler Museum, Gift of Shirley Nye in memory of Warren E. Cox   
9                          Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   
14                         Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   
15                         Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   
16                         Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   
26  Harvard Art Museums/Arthur M. Sackler Museum, Partial gift of the Walter C. Sedgwick Foundation and partial purchase through the Alpheus Hyatt Purchasing Fund   
27                                                                   Harvard Art Museums/Arthur M. Sackler Museum, Gift of Richard B. Nye in memory of Shirley Nye   
28                                                                    Harvard Art Museums/Arthur M. Sackler Museum, Gift of Shirley Nye in memory of Warren E. Cox   
34  Harvard Art Museums/Arthur M. Sackler Museum, Partial gift of the Walter C. Sedgwick Foundation and partial purchase through the Alpheus Hyatt Purchasing Fund   
35                                                                    Harvard Art Museums/Arthur M. Sackler Museum, Gift of Shirley Nye in memory of Warren E. Cox   
36                                                                    Harvard Art Museums/Arthur M. Sackler Museum, Gift of Shirley Nye in memory of Warren E. Cox   
37                                                                    Harvard Art Museums/Arthur M. Sackler Museum, Gift of Shirley Nye in memory of Warren E. Cox   
38                                                                    Harvard Art Museums/Arthur M. Sackler Museum, Gift of Shirley Nye in memory of Warren E. Cox   
39                                                                    Harvard Art Museums/Arthur M. Sackler Museum, Gift of Shirley Nye in memory of Warren E. Cox   
40                                                                    Harvard Art Museums/Arthur M. Sackler Museum, Gift of Shirley Nye in memory of Warren E. Cox   
41                        

## Take out provenance? eliminate various places such as 'Oriental Institute', 'Blitz Oriental Art' etc..

In [284]:
p_o_df =o_df[~o_df['provenance'].isna()]
p_o_df = p_o_df[p_o_df['provenance'].str.contains('Oriental')]

o_df = o_df.merge(p_o_df, how='left', indicator=True).query('_merge == "left_only"').drop(columns=['_merge'])
print(o_df.shape)

display(o_df[:20])

(126, 27)


Unnamed: 0                            copyright  \
2           98  © Estate of Joseph Janney Steinmetz   
3           18  © Estate of Joseph Janney Steinmetz   
4           14                                  NaN   
5           31                                  NaN   
30          72                                  NaN   
31          66                                  NaN   
32          94                                  NaN   
33          97                                  NaN   
34          91                                  NaN   
35          97                                  NaN   
36          43                                  NaN   
37          49                                  NaN   
38          18                                  NaN   
39          70                                  NaN   
40          42                                  NaN   
41          80                                  NaN   
42           3                                  NaN   
43           0                                  NaN   
44          77                                  NaN   
45          66                                  NaN   

                                                                                                                                 creditline  \
2   Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   
3   Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   
4   Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   
5   Harvard Art Museums/Fogg Museum, Transfer from the Carpenter Center for the Visual Arts, American Professional Photographers Collection   
30                                          Harvard Art Museums/Arthur M. Sackler Museum, Ernest B. and Helen Pratt Dane Fund for Asian Art   
31                                                                   Harvard Art Museums/Arthur M. Sackler Museum, Gift of Harry J. Denberg   
32                        Harvard Art Museums/Arthur M. Sackler Museum, Gift of The Hagop Kevorkian Foundation in memory of Hagop Kevorkian   
33                                                      Harvard Art Museums/Arthur M. Sackler Museum, Gift of Louise M. and George E. Bates   
34                                                                   Harvard Art Museums/Arthur M. Sackler Museum, Gift of Damon Mezzacappa   
35                                                               Harvard Art Museums/Arthur M. Sackler Museum, Bequest of David M. Robinson   
36                                                                Harvard Art Museums/Arthur M. Sackler Museum, Bequest of Joseph C. Hoppin   
37                                                                  Harvard Art Museums/Arthur M. Sackler Museum, Gift of the Misses Norton   
38                                                                      Harvard Art Museums/Fogg Museum, Gift of Maida and George S. Abrams   
39                                                                      Harvard Art Museums/Fogg Museum, Gift of Maida and George S. Abrams   
40                                                                             Harvard Art Museums/Fogg Museum, Bequest of Frances L. Hofer   
41                                                                             Harvard Art Museums/Fogg Museum, Bequest of Frances L. Hofer   
42                                                                        Harvard Art Museums/Fogg Museum, Bequest of Grenville L. Winthrop   
43                                                            Harvard Art Museums/Fogg Museum, Gift of Grenville L. Winthrop, Class of 1886   
44                                                            Harvard Art Museums/Fogg Museum, Gift of Grenville L. Winthrop, Class

## We are left with 126 records that should be evaluated, examples of record keyword found:
'oriental woman', 'oriental rug', 'oriental paper', 'oriental cap', 'oriental style', 'oriental dancer', etc

o_df.to_csv("oriental.csv")

## What is next?

* Add additional keywords to search
* Complete analysis of comment and desctiption fields
* Compute sentiment of comments and descriptions
* Train a language model to find specific biased text in comments and descriptions
* Look in other parts of database for bias

## Access

*  put this notebook in public github repository?
*  put keyword flagged csv's in project folder?